### PR TITLE
DIV-6180-served-by-process-server-dn-your-husband-wife-response-page

### PR DESCRIPTION
**Pull request by madjava**
# Description

https://tools.hmcts.net/jira/browse/DIV-6180

Displays a new screen when served by process server service has been issued

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by madjava** [comment]
A few changes in this ticket are as result of merging from other PR's. Main thrust of this ticket is to create a new screen when case is awaiting DN and its been served by process server service.
As there was some duplicate logic in determining if a process server service the duplicate logic has been moved to a new location to keep the `steps` classes lighter and more focused. Some other refactoring where done to reduced duplicate code/functionality.

File: helpers/petitionHelper.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  isAwaitingDecreeNisi,
+  isAwaitingPronouncementWithHearingDate,
+  isProcessServerService,
+  isServedByProcessServer,
+  isReceivedAOSFromRespondent,
+  isPetitionerRepresented,
+  isDeemedServiceApplicationGranted,
+  isDispensedServiceApplicationGranted,
+  getProcessServerReason
+};
```
Comment (madjava) [inline helpers/petitionHelper.js:83]: Helper class where some reusable functionality now lives.

File: mocks/steps/modify-session/ModifySession.content.json [hunk 2]
```diff
@@ -128,6 +148,11 @@
...
           "title": "[CY] Details of the court cases?"
         },
+        "receivedAOSfromResp": {
+          "title": "[CY] AOS Received from Respondent",
+          "yes": "[CY] Yes",
+          "no": "[CY] No"
+        },
         "respCostsReason": {
           "title": "[CY] Provide a brief explanation of why you should not pay?"
         }
```
Comment (madjava) [inline mocks/steps/modify-session/ModifySession.content.json:158]: Changes as a result of merge

File: steps/apply-for-decree-nisi/ApplyForDecreeNisi.step.js [hunk 3]
```diff
@@ -4,17 +4,17 @@ const { redirectTo } = require('@hmcts/one-per-page/flow');
...
-const Joi = require('joi');
 const i18next = require('i18next');
 const commonContent = require('common/content');
-
-const constants = {
-  no: 'no',
-  yes: 'yes',
-  deemed: 'deemed',
-  dispensed: 'dispensed'
-};
```
Comment (madjava) [inline steps/apply-for-decree-nisi/ApplyForDecreeNisi.step.js:17]: Moved to common location

File: steps/petition-progress-bar/PetitionProgressBar.step.js [hunk 4]
```diff
@@ -5,39 +5,28 @@ const idam = require('services/idam');
...
-  awaitingPronouncement: 'awaitingpronouncement',
-  awaitingClarification,
-  dnIsRefused,
-  undefendedReason: '0',
-  no: 'no',
-  yes: 'yes',
-  deemed: 'deemed',
-  dispensed: 'dispensed'
-};
-
```
Comment (madjava) [inline steps/petition-progress-bar/PetitionProgressBar.step.js:40]: Moved to common location for reuse

File: steps/petition-progress-bar/PetitionProgressBar.step.js [hunk 5]
```diff
@@ -130,14 +119,6 @@ class PetitionProgressBar extends Interstitial {
...
   }
 
-  awaitingPronouncementWithHearingDate() {
-    const isAwaitingPronouncement = this.caseState === constants
-      .awaitingPronouncement;
-    const hearingDates = get(this.case, 'hearingDate') || [];
-
-    return isAwaitingPronouncement && hearingDates.length;
-  }
-
```
Comment (madjava) [inline steps/petition-progress-bar/PetitionProgressBar.step.js:140]: Moved to helper file

File: test/unit/helpers/petitionHelper.test.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  });
+
+  it('should return true when dispensed service has been granted', () => {
+    const newSession = Object.assign({}, session);
+    newSession.case.data.serviceApplicationGranted = constants.yes;
+    newSession.case.data.serviceApplicationType = constants.dispensed;
+
+    expect(isDispensedServiceApplicationGranted(newSession.case.data)).to.equal(true);
+  });
+});
```
Comment (madjava) [inline test/unit/helpers/petitionHelper.test.js:110]: Covering test scenarios for helper class

File: mocks/steps/modify-session/ModifySession.content.json (hunk 2 above)
Comment (madjava) [inline mocks/steps/modify-session/ModifySession.content.json:158]: As a result or merge

File: steps/apply-for-decree-nisi/ApplyForDecreeNisi.html [hunk 7]
```diff
@@ -12,10 +12,16 @@
...
+
   {% elif isDispensedApproved %}
     <div class="govuk-inset-text">
       {{ content.continueBecauseOfDispensed }}
     </div>
+
+  {% elif isServedByProcessServerService %}
+    <div class="govuk-inset-text">
+      {{ content.processServerDetail }}
+    </div>
```
Comment (madjava) [inline steps/apply-for-decree-nisi/ApplyForDecreeNisi.html:24]: Logic to determine if is served by process server service to show data

File: helpers/petitionHelper.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
+const { get, toLower, isEqual, isEmpty, trim, size } = require('lodash');
```
Comment (jacmurphy) [inline helpers/petitionHelper.js:1]: fantastic addition

**Review by jacmurphy — APPROVED** [review]

File: mocks/steps/modify-session/ModifySession.content.json (hunk 2 above)
Comment (yarpo) [reply mocks/steps/modify-session/ModifySession.content.json:158]: Have you added a label to jira ticket?

Comment (yarpo) [reply mocks/steps/modify-session/ModifySession.content.json:158]: ok, I can see you have.

**Review by yarpo — APPROVED** [review]

**Pull request merged**